clevertap-core/src/main/java/com/clevertap/android/sdk/events/EventQueueManager.java

```diff
@@ -455,16 +458,18 @@ public Future<?> queueEvent(final Context context, final JSONObject event, final
             @Override
             @WorkerThread
             public Void call() {
-
+                String eventName = eventMediator.getEventName(event);
                 Location userLocation = cleverTapMetaData.getLocationFromUser();
 
+                updateLocalStore(eventName, eventType);
```

Comment:
why have we moved this from `processEvent()` to this method, making sure that we are not breaking any workflow

Reply:
We need to store data before eval